src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    @Test
+    void registerAgeTooYoung_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("YetAnotherUniqueLogin");
+        newUser.setPassword("ValidPassword123");
+        newUser.setAge(17);
+        assertThrows(IllegalArgumentException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerValidAge_Ok() {
+        User newUser = new User();
+        newUser.setLogin("YetAnotherUniqueLogin1");
+        newUser.setPassword("ValidPassword123");
+        newUser.setAge(18);
+        User registeredUser = registrationService.register(newUser);
+        assertNotNull(registeredUser);
+        assertEquals("YetAnotherUniqueLogin1", registeredUser.getLogin());
+        assertEquals("ValidPassword123", registeredUser.getPassword());
+        assertEquals(Integer.valueOf(18), registeredUser.getAge());
+    }
+
+    @Test
+    void registerTooShortLogin_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("Pasha");
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(28);
+        assertThrows(IllegalArgumentException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerValidLogin_Ok() {
+        User newUser = new User();
+        newUser.setLogin("LongEnoughLogin");
+        newUser.setPassword("ValidPassword123");
+        newUser.setAge(28);
+        User registeredUser = registrationService.register(newUser);
+        assertNotNull(registeredUser);
+        assertEquals("LongEnoughLogin", registeredUser.getLogin());
+        assertEquals("ValidPassword123", registeredUser.getPassword());
+        assertEquals(Integer.valueOf(28), registeredUser.getAge());
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        User nullUser = null;
+        assertThrows(IllegalArgumentException.class, () ->
+                registrationService.register(nullUser));
+    }
+
+    @Test
+    void registerLoginContainWhiteSpace_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("Pasha Otroda");
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(28);
+        assertThrows(IllegalArgumentException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerTooLongPassword_Ok() {
+        User newUser = new User();
+        newUser.setLogin("somelogin");
+        newUser.setPassword("ProperPassword123ProperPassword123ProperPassword123");
+        newUser.setAge(28);
+        User registeredUser = registrationService.register(newUser);
+        assertNotNull(registeredUser);
+        assertEquals("somelogin", registeredUser.getLogin());
+        assertEquals("ProperPassword123ProperPassword123ProperPassword123",
+                registeredUser.getPassword());
+        assertEquals(Integer.valueOf(28), registeredUser.getAge());
+    }
```

Comment:
You should add tests for edge cases of password length, including the minimum length that is acceptable and one that is just above the minimum length.